merino/providers/top_picks.py

```diff
@@ -133,6 +147,19 @@ def build_index(domain_list: dict[str, Any]) -> dict[str, Any]:
                 score=SCORE,
             )
 
+            # Insertion of short keys between Firefox limit of 2 and QUERY_CHAR_LIMIT - 1
+            # For similars, the values are added to the secondary index.
+            if FIREFOX_CHAR_LIMIT <= len(domain) <= (QUERY_CHAR_LIMIT - 1):
+                for chars in range(FIREFOX_CHAR_LIMIT, len(domain) + 1):
+                    short_domain_index[domain[:chars]].append(index_key)
+                if not record.get("similars"):
+                    continue
+                else:
+                    for variant in record.get("similars", []):
+                        for chars in range(FIREFOX_CHAR_LIMIT, len(variant) + 1):
+                            secondary_index[variant[:chars]].append(index_key)
+                        continue
```

Comment:
I think this can be simplified as well. Will comment in more detail in a bit.

Reply:
Yeah, I can likely just remove the if check if there is no record, I was just being explicit because this was easy to miss, proven by my missing the continue altogether and the test passing.

Reply:
I think this should just work.

```suggestion
                for variant in record.get("similars"):
                    if len(variant) >= QUERY_CHAR_LIMIT:
                        # Long variants will be indexed later into `secondary_index`
                        continue
                        
                    for chars in range(FIREFOX_CHAR_LIMIT, len(varaint) + 1):
                        short_domain_index[varinat[:chars]].append(index_key)
```

Reply:
Note that `continue` is intentionally omitted as we need the following indexer to index the long variant. Both short domain and short variants will be ignored because that `for chars in range(QUERY_CHAR_LIMIT, len(domain) + 1` will be no-op for them.

Reply:
Good call here.  I also amended tests for the longer variants as their shorter keywords of 2 - 3 chars will not trigger a lookup, just as it works with the standard domains that show at 4 char and >.